File: collection/enum/enum.go
package main

import (
	"fmt"
)

type Color int

const (
	Red Color = iota
	Green
	Blue
)

func main() {
	fmt.Println("there is no enum type in Go. Create a list of consts with iota for sequential integer")
	fmt.Println(Red, Green, Blue)
	var d Color = 0
	fmt.Println(d)

	switch d {
	case Red:
		fmt.Println("Red")
	default:
		fmt.Println("Other Color")
	}
}

File: collection/enum/enum.dart
enum Color { red, green, blue }

main(List<String> args) {
  // Color.red is an enum type
  print(Color.red);

  // Printing its name can be done with toString() as well
  print(Color.red.toString());

  // enum type has index property starting at 0
  print(Color.red.index);

  // enum type is hashable
  var m = Map();
  m[Color.red] = 1;
  print(m[Color.red]);
}
